code/game/objects/items/weapons/melee/energy.dm

```diff
@@ -51,6 +55,88 @@
 	add_fingerprint(user)
 	return
 
+/obj/item/weapon/melee/energy/handle_shield(mob/user, var/damage, atom/damage_source = null, mob/attacker = null, var/def_zone = null, var/attack_text = "the attack")
+	if(active && default_parry_check(user, attacker, damage_source) && prob(50))
+		user.visible_message("<span class='danger'>\The [user] parries [attack_text] with \the [src]!</span>")
+
+//		Disabled because lag. Immense amounts of lag.
+//		var/datum/effect/effect/system/spark_spread/spark_system = new /datum/effect/effect/system/spark_spread()

```

Comment:
I think this is actually a merge conflict, I added it back in in my sparks fix PR.